python/cucim/src/cucim/skimage/filters/thresholding.py

```diff
@@ -910,12 +912,25 @@ def threshold_triangle(image, nbins=256):
     # nbins is ignored for integer arrays
     # so, we recalculate the effective nbins.
     hist, bin_centers = histogram(image.ravel(), nbins, source_range="image")
-    nbins = len(hist)
+    if hist.size == 1:
+        # integer-valued image with constant intensity will have just 1 bin
+        return image.ravel()[0]
 
+    # In most cases, nbins is small so it is better to process hist on the CPU
+    if nbins > 100000:
+        xp = cp
+    else:
+        xp = np
+        hist = cp.asnumpy(hist)
```

Comment:
Wonder if there is something we can use from [the Array API]( https://data-apis.org/array-api/latest/ ) to make this smoother

Reply:
I think this is easy enough to follow. In practice, we will almost always take the numpy code path since default bins are 256. 

100,000 seemed a reasonable cutoff for when GPU is likely to be faster on my system but I don't think we need to try to tune this function too much as I don't think it is very heavily used.

Reply:
Yeah was more focused on maintainability and the potential to upstream more of these changes. Though need to solve that problem here. Just something to mull on